### ISSUE TITLE
Support for use of projections with find and findOne

### ISSUE THREAD

**Issue by damc-dev** [serby/save#28]
How difficult would it be to add support projections on find and findOne for the in-memory engine?

**Comment by serby** [comment]
You can add:
```
 { fields: [ '_id', 'name' ] }
```
in options. Does this do what you need?

**Comment by damc-dev** [comment]
I wasn't able to get that to work.

In the example below the id, name, email, and password is in the returned user object from find.

```
var User = require('save')('user');

User.create(
  { name: 'bob', email: 'bob@gmail.com', password: 'badpassword'},
  function(error, user) {
      console.log("User: " + user.name + " was created.");
  });

User.find(
  { email: 'bob@gmail.com' },
  { fields: ['_id', 'name'] },
  function(error, user) {
    console.log(user);
  });
```

Looking at the in-memory engine the [findByQuery](https://github.com/serby/save/blob/master/lib/memory-engine.js#L219) method doesn't seem to look for anything in the options besides sort then discards the options object.

**Comment by serby** [comment]
Ah! Your right that only works in https://github.com/serby/save-mongodb. 

Added and published in v2.1.0

https://github.com/serby/save/commit/3e4edbd830079b1a4d10a424412c063a5e28ca90
